x-pack/platform/plugins/shared/task_manager/server/lib/get_es_and_so_availability.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable, BehaviorSubject } from 'rxjs';
+import { Logger, ServiceStatusLevels, CoreStatus, IClusterClient } from '@kbn/core/server';
+import { TASK_MANAGER_INDEX } from '../constants';
+
+export interface GetElasticsearchAndSOAvailabilityOpts {
+  core$: Observable<CoreStatus>;
+  isServerless: boolean;
+  logger: Logger;
+  getClusterClient: () => Promise<IClusterClient>;
+}
+
+export function getElasticsearchAndSOAvailability({
+  core$,
+  isServerless,
+  logger,
+  getClusterClient,
+}: GetElasticsearchAndSOAvailabilityOpts): Observable<boolean> {
+  let isEsHealthy = false;
+  let isEsServiceAvailable = false;
+  let isSoServiceAvailable = false;
+  const result = new BehaviorSubject<boolean>(false);
```

Comment:
Previous code was:

```
return core$.pipe(
    map(
      ({ elasticsearch, savedObjects }) =>
        elasticsearch.level === ServiceStatusLevels.available &&
        savedObjects.level === ServiceStatusLevels.available
    ),
    distinctUntilChanged()
  );
```

I'm now creating a new BehaviorSubject and pushing values into it instead. This way I can wait for the ES health API to return before saying ES and SO are available.

I decided to move away from `distinctUntilChanged` because it is hard to write unit tests with this reducer (ex: start with false and want to assert after the the health API is called that false is still returned).